File: new1.c
#include <stdio.h>
#include <stdlib.h>

// Function to compare two integers for qsort
int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

// Function to find the minimum difference of chocolates
int min_diff_chocolates(int n, int m, int chocolates[]) {
    // Sort the chocolates array
    qsort(chocolates, n, sizeof(int), compare);
    
    // Initialize the minimum difference to a large value
    int min_diff = INT_MAX;
    
    // Traverse the sorted chocolates array to find the minimum difference
    for (int i = 0; i <= n - m; i++) {
        int current_diff = chocolates[i + m - 1] - chocolates[i];
        if (current_diff < min_diff) {
            min_diff = current_diff;
        }
    }
    return min_diff;
}
int main(){
    int n, m;
    scanf("%d %d", &m, &n);
    int *chocolates;
    chocolates = (int*)malloc(sizeof(int*));
    for(int i =0;i<n;i++){
        scanf("%d\n", &chocolates[i]);
    }
    int ans = min_diff_chocolates(n, m, chocolates);
    printf("%d ", ans);
}

File: new1.py
def min_diff_chocolates(n, m, chocolates):
    # Sort the chocolates array
    chocolates.sort()
    
    # Initialize the minimum difference to a large value
    min_diff = float('inf')
    
    # Traverse the sorted chocolates array to find the minimum difference
    for i in range(n - m + 1):
        current_diff = chocolates[i + m - 1] - chocolates[i]
        if current_diff < min_diff:
            min_diff = current_diff
    
    return min_diff

def main():
    import sys
    input = sys.stdin.read
    data = input().strip().split()
    
    # Read the values of N and M
    n = int(data[0])
    m = int(data[1])
    
    # Read the chocolates array
    chocolates = list(map(int, data[2:2 + n]))
    
    # Find and print the minimum difference
    result = min_diff_chocolates(n, m, chocolates)
    print(result)

if __name__ == "__main__":
    main()